clang/test/CodeGenSYCL/reqd-sub-group-size-device.cpp

```diff
@@ -0,0 +1,52 @@
+// RUN: %clang_cc1 -fsycl -fsycl-is-device -internal-isystem %S/Inputs -disable-llvm-passes -triple spir64-unknown-unknown-sycldevice -emit-llvm -o - %s | FileCheck %s
```

Comment:
Nit: it is not necessary to disable llvm passes in this test, I think
```suggestion
// RUN: %clang_cc1 -fsycl -fsycl-is-device -internal-isystem %S/Inputs -triple spir64-unknown-unknown-sycldevice -emit-llvm -o - %s | FileCheck %s
```

Reply:
I added -disable-llvm-passes to the test command-line to prevent inlining. so no optimization is necessary here?

Reply:
It seems the test only checks kernel metadata. I don't think that kernel can be inlined (at least on device). But I just realized that it can be possible on host.